src/chainstate/stacks/db/transactions.rs

```diff
@@ -942,6 +993,35 @@ impl StacksChainState {
                             warn!("Block compute budget exceeded: if included, this will invalidate a block"; "txid" => %tx.txid(), "cost" => %cost_after, "budget" => %budget);
                             return Err(Error::CostOverflowError(cost_before, cost_after, budget));
                         }
+                        ClarityRuntimeTxError::AnalysisError(check_error) => {
+                            if epoch_id >= StacksEpochId::Epoch21 {
+                                // in 2.1 and later, this is a permitted runtime error.  take the
+                                // fee from the payer and keep the tx.
+                                warn!("Contract-call encountered an analysis error at runtime";
+                                      "contract_name" => %contract_id,
+                                      "function_name" => %contract_call.function_name,
+                                      "function_args" => %VecDisplay(&contract_call.function_args),
+                                      "error" => %check_error);
+
+                                let receipt =
+                                    StacksTransactionReceipt::from_runtime_failure_contract_call(
+                                        tx.clone(),
+                                        total_cost,
+                                        check_error,
+                                    );
+                                return Ok(receipt);
+                            } else {
+                                // prior to 2.1, this is not permitted in a block.
+                                error!("Unexpected analysis error invalidating transaction: if included, this will invalidate a block";
```

Comment:
I think this should be `warn!` level -- `error` level logs should mean that something is _wrong_ and that the operator of the node should be trying to take action to remedy it.

Reply:
Fixed